mini-cloudinary.rb

```diff
@@ -16,56 +16,57 @@ def parse_error_to_json(code, message)
         }
         JSON[json]
     end
+
+    def build_error_array(code, message)
+        [code, parse_error_to_json(code, message)]
+    end
 end
 
 class MiniCloudinary
 
-    include ErrorHandler
-
     def initialize(path)
         @local_path = path
     end
 
-    def image_doesnt_exist?(url)
-        
-        uri = URI.parse(url)
-        request = Net::HTTP.new(uri.host, uri.port)
-        request.use_ssl = (uri.scheme == 'https')
-        response = request.request_head(uri.path)
-        response.code.to_i != STATUS_OK
+    def resize_with_black_background(image, width, height)
+        image.combine_options do |c|
+            c.extent("#{width}x#{height}")
+            c.background("black")
+            c.gravity("center")
+        end
     end
 
     def resize_image(path, output_path, width, height)
-        
-        image = MiniMagick::Image.open(path)
-        if width > image.width && height > image.height
-            image.combine_options do |c|
-                c.extent("#{width}x#{height}")
-                c.background("black")
-                c.gravity("center")
-            end
-        elsif width > image.width || height > image.height
-            image = image.resize("#{width}x#{height}")
-            image.combine_options do |c|
-                c.extent("#{width}x#{height}")
-                c.background("black")
-                c.gravity("center")
+        begin
+            image = MiniMagick::Image.open(path)
+            if width > image.width && height > image.height
```

Comment:
Hey, just checked the logic, seems that something's off with the logic.

Please check: 
https://res.cloudinary.com/miki-cloudinary/image/fetch/w_100,h_1000,c_lpad,b_black/https://asia.olympus-imaging.com/content/000107506.jpg (Cloudinary's result)

as opposed to:
https://mini-cloudinary.herokuapp.com/thumbnail?url=https://asia.olympus-imaging.com/content/000107506.jpg&width=100&height=1000 (your service result)